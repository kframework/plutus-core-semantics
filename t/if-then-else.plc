(program 0.1.0 [ [ [ { (abs a (type) (lam b (all a (type) (fun (fun (all a (type) (fun a a)) a) (fun (fun (all a (type) (fun a a)) a) a))) (lam x (fun (all a (type) (fun a a)) a) (lam y (fun (all a (type) (fun a a)) a) [ [ { b a } x ] y ]) ) ) ) (con integer) } [ (lam x_0 (all a (type) (fun (fun (all a (type) (fun a a)) a) (fun (fun (all a (type) (fun a a)) a) a))) [ [ { (builtin equalsInteger) (con 2) } [ [ { (builtin addInteger) (con 2) } [ [ { (builtin multiplyInteger) (con 2) } (con 2 ! 88) ] (con 2 ! 9) ] ] [ [ { (builtin divideInteger) (con 2) } (con 2 ! -46) ] (con 2 ! -39) ] ] ] [ (lam x_3 (all a (type) (fun (fun (all a (type) (fun a a)) a) (fun (fun (all a (type) (fun a a)) a) a))) [ [ { (builtin remainderInteger) (con 2) } (con 2 ! -65) ] (con 2 ! 3) ] ) x_0 ] ] ) [ [ { (builtin greaterThanEqualsInteger) (con 2) } [ [ { (builtin divideInteger) (con 2) } [ [ { (builtin subtractInteger) (con 2) } (con 2 ! -61) ] (con 2 ! 90) ] ] [ [ { (builtin multiplyInteger) (con 2) } (con 2 ! -25) ] (con 2 ! -82) ] ] ] [ [ { (builtin remainderInteger) (con 2) } [ [ { (builtin divideInteger) (con 2) } (con 2 ! 34) ] (con 2 ! 21) ] ] [ [ { (builtin multiplyInteger) (con 2) } (con 2 ! 83) ] (con 2 ! -89) ] ] ] ] ] [ { { (abs a (type) (abs b (type) (lam x a (lam y b x)))) (con integer) } (all a (type) (fun a a)) } [ (lam x_6 [(con integer) (con 2)] [ (lam x_9 (all a (type) (fun (fun (all a (type) (fun a a)) a) (fun (fun (all a (type) (fun a a)) a) a))) [ [ { (builtin addInteger) (con 2) } [ [ { (builtin addInteger) (con 2) } x_6 ] x_6 ] ] [ [ { (builtin addInteger) (con 2) } x_6 ] x_6 ] ] ) [ [ { (builtin equalsInteger) (con 2) } [ [ { (builtin subtractInteger) (con 2) } x_6 ] x_6 ] ] [ [ { (builtin divideInteger) (con 2) } x_6 ] x_6 ] ] ] ) [ (lam x_7 [(con size) (con 2)] [ [ { (builtin addInteger) (con 2) } [ [ { (builtin subtractInteger) (con 2) } (con 2 ! -13) ] (con 2 ! 78) ] ] [ [ { { (builtin resizeInteger) (con 2) } (con 2) } x_7 ] (con 2 ! 63) ] ] ) [ (lam x_8 [(con size) (con 2)] x_8) (con 2) ] ] ] ] ] [ { { (abs a (type) (abs b (type) (lam x a (lam y b x)))) (con integer) } (all a (type) (fun a a)) } [ (lam x_12 (all a (type) (fun (fun (all a (type) (fun a a)) a) (fun (fun (all a (type) (fun a a)) a) a))) [ [ { { (builtin resizeInteger) (con 2) } (con 2) } [ (lam x_15 (all a (type) (fun (fun (all a (type) (fun a a)) a) (fun (fun (all a (type) (fun a a)) a) a))) (con 2) ) [ [ { (builtin greaterThanEqualsInteger) (con 2) } (con 2 ! 5) ] (con 2 ! -19) ] ] ] [ [ { { (builtin resizeInteger) (con 2) } (con 2) } (con 2) ] [ [ { (builtin subtractInteger) (con 2) } (con 2 ! -7) ] (con 2 ! 88) ] ] ] ) [ [ { (builtin lessThanEqualsInteger) (con 2) } [ [ { { (builtin resizeInteger) (con 2) } (con 2) } (con 2) ] [ [ { (builtin subtractInteger) (con 2) } (con 2 ! -27) ] (con 2 ! -10) ] ] ] [ [ { { (builtin resizeInteger) (con 2) } (con 2) } (con 2) ] [ [ { (builtin divideInteger) (con 2) } (con 2 ! -1) ] (con 2 ! -87) ] ] ] ] ] ])
