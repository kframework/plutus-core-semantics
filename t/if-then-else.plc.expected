<generatedTop>
  <k>
    ( con 2 ! -95 )
  </k>
  <env>
    .Map
  </env>
  <store>
    .Map
  </store>
</generatedTop>
